eventuate-common-id/src/main/java/io/eventuate/common/id/IdGeneratorImpl.java

```diff
@@ -1,79 +1,21 @@
 package io.eventuate.common.id;
 
-import java.net.NetworkInterface;
-import java.net.SocketException;
-import java.util.Enumeration;
-
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
 public class IdGeneratorImpl implements IdGenerator {
 
-  private Logger logger = LoggerFactory.getLogger(getClass());
-  private static final long MAX_COUNTER = 1 << 16;
-
-  private Long macAddress;
-  private long currentPeriod = timeNow();
-  private long counter = 0;
-
+  public static final long SERVICE_ID_MAX_VALUE = 0x0000ffffffffffffL;
 
-  public IdGeneratorImpl() {
-    try {
-      macAddress = getMacAddress();
-      logger.debug("Mac address {}", macAddress);
-      if (macAddress == null)
-        throw new RuntimeException("Cannot find mac address");
-    } catch (SocketException e) {
-      throw new RuntimeException(e);
-    }
-  }
+  private Long serviceId;
```

Comment:
This should be `long` since it's never null.
It can also be final

Reply:
ok